modules/hostap/CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if(CONFIG_WIFI_NM_WPA_SUPPLICANT)
+
+zephyr_library()
+
+set(HOSTAP_BASE ${ZEPHYR_HOSTAP_MODULE_DIR})
+set(WIFI_NM_WPA_SUPPLICANT_BASE ${HOSTAP_BASE}/wpa_supplicant)
+set(COMMON_SRC_BASE ${HOSTAP_BASE}/src)
+
+set(CMAKE_EXE_LINKER_FLAGS "--specs=nosys.specs -lnosys")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DMISSING_SYSCALL_NAMES")
+
+zephyr_compile_definitions(
+	CONFIG_ZEPHYR
+)
+
+zephyr_include_directories(
+	${CMAKE_CURRENT_SOURCE_DIR}
+	${CMAKE_CURRENT_SOURCE_DIR}/src
+	${CMAKE_CURRENT_SOURCE_DIR}/src/utils
+	${HOSTAP_BASE}/
+	${WIFI_NM_WPA_SUPPLICANT_BASE}/
+	${COMMON_SRC_BASE}/
+)
+
+zephyr_library_compile_definitions(
+	TLS_DEFAULT_CIPHERS=\""DEFAULT:!EXP:!LOW"\"
+	CONFIG_SME
+	CONFIG_NO_CONFIG_WRITE
+	CONFIG_NO_CONFIG_BLOBS
+	CONFIG_CTRL_IFACE
+	CONFIG_CTRL_IFACE_ZEPHYR
+	CONFIG_NO_RANDOM_POOL
+	CONFIG_NO_WPA
+	CONFIG_CRYPTO_INTERNAL
```

Comment:
seems to be used only by sha1-internal.c, a file not included in the zephyr library.